packages/core/src/sdk/deliveryPromise/useDeliveryPromise.ts

```diff
@@ -0,0 +1,210 @@
+      label: selectedPickupPoint?.name ?? selectedPickupPoint?.address.street,
+      selected: !!selectedFacets.find(
+        ({ value }) => value === 'pickup-in-point'
+      ),
+      quantity: selectedPickupPoint?.totalItems,
+    }),
+    [selectedPickupPoint, selectedFacets]
+  )
+
+  const facets = useMemo(() => {
+    return !isEnabled
+      ? allFacets.filter(({ key }) => key === ShippingFacetKey)
+      : allFacets.map((facet) => {
+          if (
+            facet.key !== ShippingFacetKey ||
+            facet.__typename !== 'StoreFacetBoolean'
+          )
+            return facet
+
+          facet.values = withUniqueFacet(facet.values, allDeliveryMethodsFacet)
+          const pickupInPointFacetIndex = facet.values.findIndex(
+            (item) => item?.value === 'pickup-in-point'
+          )
+
+          // Remove old pickup `pickup in point` facet from list and search state
+          if (pickupInPointFacetIndex !== -1 && !selectedPickupPoint) {
+            const selectedShippingFacet = selectedFacets.find(
+              ({ key }) => key === ShippingFacetKey
+            )
+            if (selectedShippingFacet) {
+              const selectedPickupInPointFacets = selectedFacets.filter(
+                ({ key, value }) =>
+                  value === 'pickup-in-point' || key === PickupPointFacetKey
+              )
+
+              selectedPickupInPointFacets.length
+                ? togglePickupInPointFacet(selectedPickupInPointFacets)
+                : toggleFilterFacets([selectedShippingFacet])
+            }
+
+            // removes pickupInPointIndex from array
+            facet.values.splice(pickupInPointFacetIndex, 1)
+          }
+          // Prevent multiple `pickup in point` facet
+          else if (pickupInPointFacetIndex === -1 && selectedPickupPoint) {
+            facet.values.push(pickupInPointFacet)
+          }
+          // Replace current `pickup-in-point` facet with the updated one
+          else if (
+            facet.values[pickupInPointFacetIndex] &&
+            facet.values[pickupInPointFacetIndex]?.label !==
+              pickupInPointFacet.label
+          ) {
+            facet.values[pickupInPointFacetIndex] = pickupInPointFacet
+          }
+
+          return facet
+        })
+  }, [
+    selectedPickupPoint,
+    pickupInPointFacet,
+    toggleFilterFacets,
+    togglePickupInPointFacet,
+    allFacets,
+    selectedFacets,
+  ])
+
+  return {
+    mandatory: config.deliveryPromise.mandatory,
+    isEnabled,
+    setRegion,
+    isLoading: isLoading || loading,
+    geoCoordinates,
+    postalCode,
+    regionError,
+    clearRegionError: () => setRegionError(null),
+    pickupPoints,
+    selectedPickupPointId,
+    selectedPickupPoint,
+    pickupPointByID,
+    facets,
+  }
+}
+
+type Values = Extract<
+  Props['allFacets'][number],
+  { __typename: 'StoreFacetBoolean' }
+>['values']
+function withUniqueFacet(values: Values, facet: Values[number]) {
```

Comment:
I kinda got confused here, naming and typing. This function appends a facet we manually created? If so, what do you think about `addDeliveryMethodFacet` or similar? 

```suggestion
function addDeliveryMethodFacet(facets: Values, facet: Values) {
```

Reply:
This is adds a value to a Filter_FacetsFragment values without duplicating it. 
I changed the typing, see if you prefer this way